### PR TITLE
LSD rework

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
[FEATURE] [GAMEMODES] [CHEMISTRY]
## About the PR 
Makes LSD hallucinations a little more realistic:
- Hallucinations now pull from a much larger pool of beasties including changeling critters
- Hallucinations now move and attack more realistically
- Audio hallucinations now pull from a much larger pool, and can play multiple times depending on the sound (eg gunshots and attack sounds play multiple times with their respective attack delays)
- The colour shifting part is now slightly delayed so the first few hallucinations are much more likely to be mistaken for reality
- The chances of some effects have been adjusted slightly
- People around you (including yourself) will now occasionally appear to transform into critters

Turns out byond fixes the movement states issue if you set `override` on an image anyway so that all works fine now, all that remains is a scuffed commit history.

## Why's this needed? 
LSD is currently just an annoyance, none of the hallucinations are at all realistic. This aims to make it actually useful for changelings and other antags to cause confusion and cover up their activities. For example some common vampire and arcfiend sound effects have been included.

## Changelog 

```changelog
(u)LeahTheTech
(+)LSD hallucinations are now a little more realistic.
```

### CODE REVIEW

**Comment by Laboredih123** [comment]
relative pathing wahh

**Comment by TobleroneSwordfish** [comment]
> relative pathing wahh

I wanted it to be as low effort as possible to add more subtypes, but could switch to absolute pathing I spose.

File: code/modules/chemistry/Reagents-Drugs.dm [hunk 1]
```diff
@@ -289,53 +289,89 @@ datum
...
+					var/turf/mob_turf = get_turf(M)
+					if (mob_turf)
+						origin = locate(mob_turf.x + rand(-10,10), mob_turf.y + rand(-10,10), mob_turf.z)
+					//wacky loosely typed code ahead
+					var/datum/hallucinated_sound/chosen = pick(src.halluc_sounds)
+					if (istype(chosen)) //it's a datum
+						SPAWN(0)
+							for (var/i = 1 to rand(chosen.min_count, chosen.max_count))
+								M.playsound_local(origin, chosen.path, 100, 1, pitch = chosen.pitch)
+								sleep(chosen.delay)
```
Comment (Tarmunora) [inline code/modules/chemistry/Reagents-Drugs.dm:369]: wouldn't it make more sense for this to be a proc on the `hallucinated_sound` datum?

**Review by Tarmunora — CHANGES_REQUESTED** [review]

**Comment by muzcl** [comment]
a video to show the effects would be nice

**Comment by TobleroneSwordfish** [comment]
Okay I got movement states working, but the implementation is reasonably cursed, see "Movement states hack" commit.

**Comment by TobleroneSwordfish** [comment]
> a video to show the effects would be nice

The effects are so random it's very hard to show them off in a short clip, if you want to test it you can always checkout the PR.

**Review by Tarmunora — APPROVED** [review]

**Comment by Crackaduck** [comment]
amazing, its going to be so fun having people actually believe in their trips

**Pull request merged**